### PR TITLE
[EDA-689] Remove join company step

### PR DESCRIPTION
**Pull request by nicolasnunez-eb**
CompanySelectorView removed, now when a supplier click on the link and login he is automatically related with the company

### CODE REVIEW

File: supplier_app/views.py [hunk 1]
```diff
@@ -270,13 +249,24 @@ def company_invite(request):
...
 @permission_required('users_app.supplier_role', raise_exception=True)
-def company_join(request, token):
-    user = request.user
-    companyuniquetoken = get_object_or_404(CompanyUniqueToken, token=token)
-    company = companyuniquetoken.company
-    CompanyUserPermission.objects.create(user=user, company=company)
-    companyuniquetoken.delete()
-    return redirect('supplier-home')
+def company_join(request, *args, **kwargs):
+    companyuniquetoken = _get_companyuniquetoken_from_token(kwargs['token'])
```
Comment (leandro-eb) [inline supplier_app/views.py]: Please replace `companyuniquetoken` by `company_unique_token`.

File: supplier_app/views.py [hunk 2]
```diff
@@ -270,13 +249,24 @@ def company_invite(request):
...
+        company = companyuniquetoken.company
+        CompanyUserPermission.objects.create(user=user, company=company)
+        companyuniquetoken.delete()
+        return redirect('supplier-home')
+    return HttpResponseRedirect(Http404)
+
+
+def _token_is_valid(companyuniquetoken):
+    if companyuniquetoken and not companyuniquetoken.is_token_expired:
+        return True
```
Comment (leandro-eb) [inline supplier_app/views.py]: You could also return that if condition. Example:
```
def _token_is_valid(company_unique_token):
    return company_unique_token and not company_unique_token.is_token_expired
```

File: supplier_app/views.py (hunk 1 above)
Comment (nicolasnunez-eb) [reply supplier_app/views.py]: done

File: supplier_app/views.py (hunk 2 above)
Comment (nicolasnunez-eb) [reply supplier_app/views.py]: done

**Pull request merged**